src/org/zaproxy/zap/extension/fieldenumeration/ExtensionFieldEnumeration.java

```diff
@@ -0,0 +1,87 @@
+package org.zaproxy.zap.extension.fieldenumeration;
+
+import java.awt.CardLayout;
+import java.awt.Font;
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.text.MessageFormat;
+
+import javax.swing.ImageIcon;
+import javax.swing.JTextPane;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.utils.FontUtils;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+public class ExtensionFieldEnumeration extends ExtensionAdaptor {
+
+	private PopupMenuField popupMenuField = null;
+        private FieldEnumeration fieldEnumeration = null;
+	public static final String NAME = "ExtensionFieldEnumeration";
+	protected static final String PREFIX = "fieldenumeration";
+	
+	public ExtensionFieldEnumeration() {
+        	super(NAME);
+	}
+	
+	@Override
+	public void hook(ExtensionHook extensionHook) {
+	    super.hook(extensionHook);
+	    
+	    if (getView() != null) {
+	        extensionHook.getHookMenu().addPopupMenuItem(getPopupMenuField());
+	    }
+
+	}
+
+
+	private PopupMenuField getPopupMenuField() {
+        if (popupMenuField == null) {
```

Comment:
The file seems to contain mixed indentation characters. This should be standardized throughout.